Refactor hover doc comment extraction

wiyota opened the pull request:
Implement multi-strategy approach with three methods for finding documentation: definitions lookup, direct node search, and pattern matching. Add debug logging and improve handling of various node types including function declarations, let bindings, and object fields.

These changes should greatly improve the development experience, as doc comments will now appear in class methods, etc.

Kamirus commented on src/server/handlers.ts:
```diff
@@ -1407,36 +1407,177 @@ export const addHandlers = (connection: Connection, redirectConsole = true) => {
+            let depth = 0;
+            const maxDepth = 4;
+
+            // First check if current node has doc
+            if (docNode.doc) {
+                return docNode.doc;
+            }
```
redundant, this check is the same in the while-loop

Kamirus commented on src/server/handlers.ts:
```diff
@@ -1407,36 +1407,177 @@ export const addHandlers = (connection: Connection, redirectConsole = true) => {
-                    const doc = asNode(docNode.args?.[0])?.doc;
-                    if (doc) {
-                        docs.push(doc);
+                    const firstChild = asNode(docNode.args?.[0]);
+                    if (firstChild?.doc) {
+                        return firstChild.doc;
```
Why change? is the `const doc = asNode(docNode.args?.[0])?.doc;` not good enough?

wiyota replied:
The intent was to emphasize that it is the first child, but it is certainly not particularly necessary, so I will undo the expression.

Kamirus reviewed (commented):
Thanks for your contribution! @rvanasa would probably want to look at this PR next week, he's currently on vacation.
@wiyota would it be possible to cover your new functionality with tests? I see there are more cases where docs are resolved

rvanasa reviewed (commented):
Thank you for the contribution! This will be useful, although I found a few possible edge cases with this logic. 

For instance:

```motoko
/// Class-level doc comment
class Example() {
    public func method() {};
};
```

With the new logic, hovering over `method` shows the doc comments for the enclosing class. Is this intentional? If so, maybe we could show the relevant line (in this case, `class Example() {`) in a code block above the doc comment for clarity.

wiyota commented:
I've discontinued the previous implementation and instead enhanced the original approach to ensure more reliable document retrieval.
While I’ve added processing for specific nodes, I’ve encountered cases where documents appear but type hints fail to display. I plan to integrate and strengthen these processing functions into individual methods within navigation.ts.

wiyota commented:
Implemented significant functionality improvements and organized/added tests.

The changes include:

- Prioritizing `ID` nodes primarily when detecting nodes during hover events to reduce scope bloat (resulting in hover scopes no longer spanning multiple lines)
- When an `ID` node lacks a `type`, it now infers the pattern from its ancestor nodes and attempts to locate the `type`
- Treats tag names within Variant declarations as pseudo-nodes and displays them when they have associated types
- Displays members of Records when they have types
- Adds type and doc comment display functionality for Class declarations
- Retrieves and displays external module types from import statements
- Treats `module` and `actor` keywords as pseudo-nodes and displays their types and doc comments
- Modified the interpretation of Motoko code blocks in doc comments—including those marked with annotations like `include=import` and those without language identifiers—to treat them as standard Motoko code blocks, enabling proper code highlighting
- Removed the original tests added in this PR and replaced them with more comprehensive "hover" tests
